### ISSUE TITLE
Dropdown component updates

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#718]
Several changes were made to the Dropdown component in core (https://github.com/patternfly/patternfly-next/pull/750) that should be reflected here:

- **Dropdown Select** - This is a new variation of the Dropdown that supports the use case of making a selection. The language selector on the login page is a good example of this component being used. Some things to note about this component:
  - The menu items are always `<button>` elements and `aria-pressed` is always present on all of them. The menu item that's currently selected has `aria-pressed="true"`. Otherwise, it should be set to `"false"`
  - An additional `<span>` is included before the `<button>` toggle to provide a visually hidden but screen reader accessible label for the menu. Without seeing this component used in context of different design patterns, I'm not sure if this would be optional or required for accessibility. Maybe we make it optional and present a warning?
  - the menu `<div>` is now a `<ul>` with `<li>` elements that wrap each `<a>` or `<button>`. The `aria-labelledby` that was on the `<div>` is now on the `<ul>` 
  - The `aria-labelledby` attributes are different for this variation. 
    - The button is labelled by the hidden label and itself. 
    - The menu is labelled by the hidden label.

To be implemented in a separate issue/PR:
- **Dropdown Multi-Select** - This is another variation of the Dropdown component that we show in core. There are still some moving parts for this one in core, so we'll add this in a separate issue.

**Comment by jgiardino** [comment]
After further discussion and reviewing wireframe designs from @mcarrano, which are based on the following [example in PF3](https://patternfly-react.surge.sh/patternfly-3/?selectedKind=patternfly-react%2FForms%20and%20Controls%2FFilter&selectedStory=Filter&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs), I wanted to capture some updates to html we need and the expected interaction behavior.

![image](https://user-images.githubusercontent.com/21063328/46823709-e8ba2a00-cd5c-11e8-9716-187d9a1f7609.png)

@mcarrano @tlabaj - let me know if this doesn't match what you were thinking, and if this should be captured in a separate issue.

**Updates needed to the html**

In core, we currently have the following:
```
<div class="pf-c-dropdown pf-m-expanded">
  <span id="dropdown-select-example-expanded-label" hidden>Select a fruit</span>
  <button id="dropdown-select-example-expanded-button" class="pf-c-dropdown__toggle" aria-labelledby="dropdown-select-example-expanded-label dropdown-select-example-expanded-button" aria-expanded="true">
    Choose one
    <i class="fas fa-caret-down pf-c-dropdown__toggle-icon" aria-hidden="true"></i>
  </button>
```
In this implementation, we have a hidden label and a placeholder that are different text strings. But instead, these strings should be the same. 

For this case where we have placeholder text that initially displays, we should **make the following changes to what's shown in core**:
- remove `aria-labelledby` from the button, since the button text is the same as the hidden label
```
<div class="pf-c-dropdown pf-m-expanded">
  <span id="dropdown-select-example-expanded-label" hidden>Select a fruit</span>
  <button id="dropdown-select-example-expanded-button" class="pf-c-dropdown__toggle" aria-expanded="true">
    Select a fruit
    <i class="fas fa-caret-down pf-c-dropdown__toggle-icon" aria-hidden="true"></i>
  </button>
```

For the html where an option is selected, either as a default option or as a result of user selection, we would continue to use the html as it's implemented in core for this example:

![image](https://user-images.githubusercontent.com/21063328/46829215-2b82fe80-cd6b-11e8-8181-751c281e6492.png)


**Expected interaction behavior**

- For cases where an option is selected by default, as shown here
  ![image](https://user-images.githubusercontent.com/21063328/46828889-4143f400-cd6a-11e8-91f5-4be603a596b5.png)
  - Selecting the option that's currently selected does not change the selection. The current option remains selected (i.e. `aria-pressed` will still equal `true`)
  - Selecting an option that's not currently selected will:
    - update the button label to display the text string from the selected option
    - toggle value of aria-pressed from true to false for the previously selected option 
    - toggle value of aria-pressed from false to true for the newly selected option
- For cases where an option is not selected by default, as shown here
  ![image](https://user-images.githubusercontent.com/21063328/46828957-6a648480-cd6a-11e8-8262-aa9137b99698.png)
  - The behavior above still applies
  - When an option is selected, it cannot be unselected by using the select menu. This use case would be used in combination with other design patterns, like filtering, where another component includes the action to clear the filter setting, which results in restoring the default display of the select menu.

**Comment by mcarrano** [comment]
This all sounds correct to me @jgiardino . Thanks for documenting.

**Comment by amarie401** [comment]
@tlabaj is this a continuation of this issue? https://github.com/patternfly/patternfly-react/issues/787

**Comment by tlabaj** [comment]
@amarie401 I split out the accessibility updates into a separate issue since they were ready to PR and there are still some discussions going on about the Dropdown Select.

**Comment by LHinson** [comment]
Putting this on hold until the core implementation is worked out for custom select. See: https://github.com/patternfly/patternfly-next/issues/651

Once that is complete, we will revisit this issue.

**Comment by tlabaj** [comment]
This issues will be covered by the Custom Select Component issue #1206
